src/service.cc

```diff
@@ -424,9 +457,10 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
         Status s_i(statuses[i].get());
         // always retrieve the result BCF record, if any, to ensure we'll free
         // the memory it takes ASAP
-        shared_ptr<bcf1_t> bcf_i = move(results[i].first);
-        assert(!results[i].first);
-        consolidated_loss losses_for_site = move(results[i].second);
+        shared_ptr<bcf1_t> bcf_i = move(std::get<0>(results[i]));
+        assert(std::get<0>(results[i]) == nullptr);
+        consolidated_loss losses_for_site = move(std::get<1>(results[i]));
+        shared_ptr<string> residual_rec =  std::get<2>(results[i]);
```

Comment:
Ah, one subtlety here: if you don't wipe out the residuals YAML here, it will all stay in memory (ALL the sites). This is the reason why we `move` the two other elements which we should also do for the third.